src/ExcelProvider.Runtime/ExcelProvider.Runtime.fs

```diff
@@ -329,15 +338,19 @@ type Row(documentId, sheetname, rowIndex, getCellValue: int -> int -> obj, colum
             |> Seq.map (fun kvp -> kvp.Key)
             |> Seq.tryFind (fun header -> String.Equals(header, columnName, StringComparison.OrdinalIgnoreCase))
             |> function
-                | Some header -> sprintf "ExcelProvider: Column \"%s\" was not found. Did you mean \"%s\"?" columnName header
+                | Some header ->
+                    sprintf "ExcelProvider: Column \"%s\" was not found. Did you mean \"%s\"?" columnName header
                 | None -> sprintf "ExcelProvider: Column \"%s\" was not found." columnName
             |> failwith
 
     member this.TryGetValue<'a> (columnIndex: int) columnName =
         let value = this.GetValue columnIndex
 
         try
-            value :?> 'a
+            match value with
+            | null when typeof<'a> = typeof<string> -> value :?> 'a
+            | _ when typeof<'a> = typeof<string> -> (box (string value)) :?> 'a
+            | _ -> value :?> 'a
```

Comment:
This is pretty much it. 

The first line that propagates the null through is to protect that expected behavior. 

On the second line, I'll admit that I haven't played with casting as much in F#, but a lot of this is just convincing the compiler that `'a` doesn't have to be constrained to string here.
Our downcast should always work since we know `'a` is `string` from the guard, and we just wrapped the known string value with `box`.